handlers/callback.go

```diff
@@ -209,15 +209,17 @@ func HandlePayPalCallback(externalPaymentSvc service.PaymentProviderService) htt
 			return
 		}
 		captureStatus := response.PurchaseUnits[0].Payments.Captures[0].Status
-		if !strings.EqualFold(captureStatus, "completed") {
-			log.ErrorR(req, fmt.Errorf("error - paypal payment status not completed, status is: [%s]", captureStatus))
-			w.WriteHeader(http.StatusInternalServerError)
-			return
+		switch captureStatus {
+		case "COMPLETED":
+			paymentSession.Status = service.Paid.String()
+		case "DECLINED":
+			paymentSession.Status = service.NoFunds.String()
+		case "FAILED":
+			paymentSession.Status = service.Failed.String()
+		default:
+			paymentSession.Status = service.Failed.String()
 		}
-		// TODO handle unsuccessful capture statuses
 
-		// Set the status of the payment
-		paymentSession.Status = "paid"
 		// To match the format time is saved to mongo, e.g. "2018-11-22T08:39:16.782Z", truncate the time
 		paymentSession.CompletedAt = time.Now().Truncate(time.Millisecond)
```

Comment:
Should CompletedAt be omitted if status is not completed?

Reply:
@jphillips2121 Does any service depend on the `CompletedAt` field even in failure scenarios?

Reply:
Not that I'm aware of, however GovPay always adds in a `CompletedAt` currently regardless of status. It might be handy to keep it in for support to see when a payment was marked as ended. Technically the payment session is "completed" just not necessarily with a successful payment